### ISSUE TITLE
Windows detecting As "Virus/Malware"

### ISSUE THREAD

**Issue by Sachin247c** [efroemling/ballistica#413]
OS - Windows 11 with latest update done! 
Problem - Windows defender (inbuilt) detecting "Bombsquad Windows 1.6.12" as a Virus           ( Refrence pic 1)
Not sure if that problem causing due to windows 11 or some kind of unrecognised "code" being in the package...

Previously I had "Bombsquad Windows 1.6.4" working Flawlessly. 
But now downloading new version (Bombsquad Windows 1.6.12) from chrome the Defender deletes it instantly! Not even telling what the problem is! 

But downloading if using "IDM" the windows defender is not deleting that instantly but showing            (Refrence pic 2)

"AndroidOS/ZkarletFlash" Maybe i am not Smart enough to figure it out !! 
I know i can "Exclude" this files being detect as a Virus but still _why_ i have to for a game this well made?!

![Screenshot 2022-06-01 122102](https://user-images.githubusercontent.com/67496924/171346001-75d42d61-6119-418e-af09-8b3773183a17.png)
![Screenshot 2022-06-01 122611](https://user-images.githubusercontent.com/67496924/171345995-3146dd5c-7539-4c4b-b0ea-c799ed16a242.png)

**Comment by Sachin247c** [comment]
> This is basically known issue (#195, #349, #357). Though it could be possible to look where exactly scanners get confused and fix that place if possible.

> This is basically known issue (#195, #349, #357). Though it could be possible to look where exactly scanners get confused and fix that place if possible.

Yes, I know this is a "known issue" and reported previously! But isn't it annoying, and why does it have to be treated like a Virus/Malware when in previous versions of the game, it was never so! Can we just not fix that? And as I have seen _nobody_ actually said about "Androidos/zkarlet......" I just mentioned it so one can pinpoint the problem in the game! well i am no Dev./coder so i don't know how to fix it but let the pros do the work.

**Comment by imayushsaini** [comment]
I just tried a simple test , I downloaded 1.6.12 build and put .zip and extracted files in same folder , with Microsoft's defender I scanned that folder and it gave me same virus report "AndroidOS/ZkarletFlash" only on that .zip not on extracted files .

on Test 2 , I put original downloaded .zip and  .zip made with windows default compressor on same extracted files . 
in this test virus detected only on original .zip , and auto remover removed only official.zip . (SHA was different for both zips).

If someone else can perform same test on their system , then we can confirm problem is with compression method only .

**Comment by imayushsaini** [comment]
![image](https://user-images.githubusercontent.com/36878972/171453865-42caa648-7ce4-4713-9dfb-98e3970a3395.png)

![image](https://user-images.githubusercontent.com/36878972/171453930-5d58f8c3-5ef6-4fe9-8e0c-8a88d177cc46.png)

release build vs build on my system

**Comment by Sachin247c** [comment]
> I just tried a simple test , I downloaded 1.6.12 build and put .zip and extracted files in same folder , with Microsoft's defender I scanned that folder and it gave me same virus report "AndroidOS/ZkarletFlash" only on that .zip not on extracted files .
> 
> on Test 2 , I put original downloaded .zip and .zip made with windows default compressor on same extracted files . in this test virus detected only on original .zip , and auto remover removed only official.zip . (SHA was different for both zips).
> 
> If someone else can perform same test on their system , then we can confirm problem is with compression method only .

Did the same! (1.Checking the main downloaded file, 2.Decompressing it using WinRar, 3.Again compressing it using windows compressor. 

My results - 1. As usual the main downloaded file "Bombsquad Windows 1.6.12" Showed the _same_ type of trojans as yours on www.virustotal.com
               
                    2.And after compressing "Bombsquad Windows 1.6.12" to "Bombsquad Windows 1.6.12(Compressed Again)" and checking it on virustotal in my case showed **no difference** at all!

And also checked the same using "Windows security" the results were **Also same** Showed "AndroidOS/ZkarletFlash"

**Solution** - 1. Eric can run security check on his whole system.
                       2. Someone other should _make a prebuild_ on their own system and check the same (ps- idk how to make a prebuild lol)  And if the results are positive then Eric's system is buggy and if not then SOME other thing is interpreting with the build.


![Screenshot 2022-06-02 093319](https://user-images.githubusercontent.com/67496924/171551216-7fce2542-b93e-4370-a8ba-70ce99044cc9.png)

**Comment by efroemling** [comment]
> official game (1.6.12) Vs debug build (1.7.0) on my system Vs release build (1.7.0) on my system

Interesting. The 1.7 builds you mention: are those from the 'prefab' build targets? Or did you compile them yourself? (the `windows_debug`, `windows_release` targets).

**Comment by imayushsaini** [comment]
> > official game (1.6.12) Vs debug build (1.7.0) on my system Vs release build (1.7.0) on my system
> 
> Interesting. The 1.7 builds you mention: are those from the 'prefab' build targets? Or did you compile them yourself? (the `windows_debug`, `windows_release` targets).

I compiled from source , 

for better comparison and tests I am attaching my builds 
1.  1.6.12 (20551) commit - 32caf4a
2.  1.7.0 (python3.10) commit -  9b27b54
build from source on fresh clone
https://www.mediafire.com/folder/p3zhfpxpdmn9y/my_builds

**Comment by efroemling** [comment]
I just ran the latest BallisticaCore prefab binaries through virustotal.com and got only 2 flags for the release binary and 1 for the debug binary. These binaries are built on the exact same machine as the public BombSquad test builds which gets about 10 flags. Basically the only difference is that the word 'BallisticaCore' is replaced with 'BombSquad' in those builds. Maybe it's the word 'Bomb' 🤣.

This is all so frustratingly inconsistent 😑.

I'll get BombSquad test builds up for the new 1.7 stuff within the next few days and I'm curious what the totals will be for them.
